api/routes/Auth.js

```diff
@@ -0,0 +1,201 @@
+                { pagesPrinted: 0 },
+                function(error, result) {
+                  // if (error) return res.sendStatus(INTERNAL_SERVER_ERROR)
+                  if (error) {
+                    logger.log(
+                      'Bad request while trying to reset pageCount to 0 for ' +
+                        user.email
+                    );
+                  }
+
+                  if (result.nModified < 1) {
+                    return logger.log(
+                      `Cannot update ${user.email}, email was not found.`
+                    );
+                  }
+                }
+              );
+            }
+
+            // Include fields from the User model that should
+            // be passed to the JSON Web Token (JWT)
+            const userToBeSigned = {
+              firstName: user.firstName,
+              lastName: user.lastName,
+              email: user.email,
+              accessLevel: user.accessLevel,
+              pagesPrinted: user.pagesPrinted
+            };
+            const token = jwt.sign(
+              userToBeSigned, config.secretKey, jwtOptions
+            );
+            res.status(OK).send({ token: 'JWT ' + token });
+          } else {
+            // Unauthorized
+            logger.log('User/pass doesn\'t match our records: ', user);
+            res.status(UNAUTHORIZED).send({
+              message: 'Username or password does not match our records.'
+            });
+          }
+        });
+      }
+    }
+  );
+});
+
+// Edit/Update a member record
+router.post('/setEmailToVerified', (req, res) => {
+  const query = { email: req.body.email };
+
+  User.updateOne(query, { emailVerified: true }, function(error, result) {
+    if (error) {
+      const info = {
+        userEmail: req.body.email,
+        errorTime: new Date(),
+        apiEndpoint: 'user/setEmailToVerified',
+        errorDescription: error
+      };
+      addErrorLog(info);
+      res.status(BAD_REQUEST).send({ message: 'Bad Request.' });
+    }
+
+    if (result.nModified < 1) {
+      return res
+        .status(NOT_FOUND)
+        .send({ message: `${req.body.queryEmail} not found.` });
+    }
+
+    return res.status(OK).send({
+      message: `${req.body.queryEmail} was updated.`
+    });
+  });
+});
+
+// Verifies the users session if they have an active jwtToken.
+// Used on the inital load of root '/'
+// Returns the name and accesslevel of the user w/ the given access token
+router.post('/verify', function(req, res) {
+  console.log(checkIfTokenValid(req));
+  if (!checkIfTokenSent(req)) {
+    return res.sendStatus(UNAUTHORIZED);
+  }
+  // else if (!checkIfTokenValid(req)) {
+  //   return res.sendStatus(UNAUTHORIZED);
+  // }
+  // else{
+  //   return res.status(OK).send(checkIfTokenValid(req));
+  // }
+
+  const decoded = checkIfTokenValid(req);
+  if (!decoded) {
+    res.sendStatus(UNAUTHORIZED);
+  } else {
+    res.status(OK).send(decoded);
+  }
+});
+
```

Comment:
what do this do

Reply:
just noting this down bc we have comments on some of the router.post() calls, and then there are none so it feels a bit inconsistent -- we should come back to this when we do documentation

Reply:
we need docs on api stuff, you're right